tests/cpp/test_alias.cpp

```diff
@@ -1545,4 +1545,38 @@ TEST_F(AliasTest, Issue2664) {
       __FILE__);
 }
 
+TEST_F(AliasTest, AliasOutputNoSegmentation) {
+  // testing a complete fusion
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  const std::vector<int64_t> in_shape({2, 3, 4});
+
+  TensorView* in = makeContigConcreteTensor(in_shape);
+  fusion->addInput(in);
+  TensorView* out = add(in, IrBuilder::create<Val>(3.141));
+  fusion->addOutput(out);
+  // this is an inplace update and shouldn't be segmented into its own kernel by alias analysis
+  TensorView* update_input = set(out);
+  fusion->aliasOutputToInput(update_input, in, AllocationType::ReuseBuffer);
+
+  FusionExecutorCache executor_cache(std::move(fusion));
+  at::Tensor in_tensor =
+      at::randn(in_shape, at::dtype(at::kFloat).device(at::kCUDA, 0));
+  at::Tensor original_tensor = in_tensor.clone();
+  std::vector<at::Tensor> out_tensors =
+      executor_cache.runFusionWithInputs({in_tensor});
+  ASSERT_EQ(out_tensors.size(), 1);
+
+  // Verify inplace update
+  EXPECT_TRUE(out_tensors[0].equal(in_tensor));
+  // Verify no segmentation
+  EXPECT_FALSE(executor_cache.getMostRecentKernelRuntime()->isSegmented())
```

Comment:
https://testing.googleblog.com/2013/08/testing-on-toilet-test-behavior-not.html

I think what we really care about is that the input is correctly updated. Whether the fusion is segmented is an implementation detail that's not observable to the user. Wdyt?

Reply:
I felt like that I needed something to justify the PR.

If we just care about correctness, there's nothing wrong with ToT code as-is. But segmentation could lead to regression (compilation time as well as execution time), which is arguably an observable thing to user. :)

If you think the test is a bit too explicit. I'm more than happy to remove it. I hope to keep it in case we refactor this pass later and might hit the same issue again.

Reply:
> If we just care about correctness, there's nothing wrong with ToT code as-is. But segmentation could lead to regression (compilation time as well as execution time), which is arguably an observable thing to user. :)

Got it! Then it's certainly fine to keep the segmentation check here.